test/built-ins/RegExp/S15.10.2.10_A5.1_T1.js

```diff
@@ -11,12 +11,8 @@ description: "Tested string is \"~`!@#$%^&*()-+={[}]|\\\\:;'<,>./?\" + '\"'"
 
 //CHECK#1
 var non_ident = "~`!@#$%^&*()-+={[}]|\\:;'<,>./?" + '"';
-var k = -1;
-do {
-   k++;
-   var arr = new RegExp("\\" + non_ident[k], "g").exec(non_ident);   
-} while ((arr !== null) && (arr[0] === non_ident[k]))
-
-if (non_ident.length !== k) {
-   $ERROR('#1: IdentityEscapeSequence :: SourceCharacter but not IdentifierPart');
+for (var k = 0; k < non_ident.length; ++k) {
+  var arr = new RegExp("\\" + non_ident[k], "g").exec(non_ident);
+  assert(arr !== null, "No match for character: " + non_ident[k]);
```

Comment:
I think it would be better to use `assert.notSameValue(arr, null, ...)` here. We would get more information in the execution log this way.